irc_send.c

```diff
@@ -213,13 +213,7 @@ void irc_send_names(irc_channel_t *ic)
 			*namelist = 0;
 		}
 
-		if (icu->flags & IRC_CHANNEL_USER_OP) {
-			strcat(namelist, "@");
-		} else if (icu->flags & IRC_CHANNEL_USER_HALFOP) {
-			strcat(namelist, "%");
-		} else if (icu->flags & IRC_CHANNEL_USER_VOICE) {
-			strcat(namelist, "+");
-		}
+		namelist[strlen(namelist)] = irc_channel_user_get_prefix(icu);
```

Comment:
Actually https://developer.gnome.org/glib/stable/glib-String-Utility-Functions.html is not about GStrings. But it doesn't even have a char-append function so never mind.

strcat() should be safe since I do assume it would add a new trailing \0 which is what I was missing here. But as said I think this should not be necessary. (Though I don't remember whether the function does other naughty things that don't do that.)